source/common/http/filter_manager.cc

```diff
@@ -1253,14 +1255,18 @@ void ActiveStreamDecoderFilter::setDecoderBufferLimit(uint32_t limit) {
 
 uint32_t ActiveStreamDecoderFilter::decoderBufferLimit() { return parent_.buffer_limit_; }
 
-bool ActiveStreamDecoderFilter::recreateStream() {
+bool ActiveStreamDecoderFilter::recreateStream(const ResponseHeaderMap* headers) {
   // Because the filter's and the HCM view of if the stream has a body and if
   // the stream is complete may differ, re-check bytesReceived() to make sure
   // there was no body from the HCM's point of view.
   if (!complete() || parent_.stream_info_.bytesReceived() != 0) {
     return false;
   }
 
+  if (headers != nullptr) {
```

Comment:
Q: is it better to call chargeStats with an empty ResponseHeaderMap when headers is nullptr? At least to can charge other non-response-header-dependent stats.

Reply:
This comment is now irrelevant based on my newest comment about skipping chargeStats